neurom/core/_soma.py

```diff
@@ -43,24 +42,38 @@ class Soma:
     and provides iterator access to them.
     """
 
-    def __init__(self, points):
-        """Initialize a Soma object."""
-        self._points = points
+    def __init__(self, morphio_soma):
+        """Constructor.
+
+        Args:
+            morphio_soma (morphio.Soma): instance of soma of MorphIO class
+        """
+        self._morphio_soma = morphio_soma
+        # this radius is used only for `volume` method, please avoid using it for anything else.
         self.radius = 0
 
     @property
     def center(self):
         """Obtain the center from the first stored point."""
-        return self._points[0][COLS.XYZ]
+        return self._morphio_soma.points[0]
 
     def iter(self):
         """Iterator to soma contents."""
-        return iter(self._points)
+        return iter(self.points)
 
     @property
     def points(self):
         """Get the set of (x, y, z, r) points this soma."""
-        return self._points[:, COLS.XYZR]
+        return np.concatenate((self._morphio_soma.points,
+                               self._morphio_soma.diameters[:, np.newaxis] / 2.),
+                              axis=1)
+
+    @points.setter
+    def points(self, value):
+        """Set the points."""
+        value = np.asarray(value)
+        self._morphio_soma.points = np.copy(value[:, COLS.XYZ])
+        self._morphio_soma.diameters = np.copy(value[:, COLS.R]) * 2
 
     @property
     def volume(self):
```

Comment:
not sure where self.radius is updated, is it when the soma is constructed, there is soma.radius = 10?

Reply:
Logic of `self.radius` stayed the same as before. Each Soma class initialized it with corresponding value.